part_2/second_split/method_v2.R

```diff
@@ -0,0 +1,398 @@
+upper_clean_gcu <- quantile(data_agg_sum_trans_log$gcu_seconds, 0.999)
+lower_clean_gcu <- quantile(data_agg_sum_trans_log$gcu_seconds, 0.001)
+upper_clean_mem <- quantile(data_agg_sum_trans_log$memory_gib_seconds, 0.995)
+lower_clean_mem <- quantile(data_agg_sum_trans_log$memory_gib_seconds, 0.005)
+
+ind_clean_gcu <- which(data_agg_sum_trans_log$gcu_seconds > upper_clean_gcu | 
+                         data_agg_sum_trans_log$gcu_seconds < lower_clean_gcu)
+ind_clean_mem <- which(data_agg_sum_trans_log$memory_gib_seconds > upper_clean_mem | 
+                         data_agg_sum_trans_log$memory_gib_seconds < lower_clean_mem)
+
+missingdata <- as.data.frame(c(ind_clean_gcu, ind_clean_mem))
+colnames(missingdata) <- "index"
+missingdata$features <- rep(c("gcu_seconds", "memory_gib_seconds"), c(length(ind_clean_gcu), length(ind_clean_mem)))
+missingdata$original_data <- 0
+missingdata$original_data[1:length(ind_clean_gcu)] <- data_agg_sum_trans_log$gcu_seconds[ind_clean_gcu]
+missingdata$original_data[(length(ind_clean_gcu)+1):nrow(missingdata)] <- data_agg_sum_trans_log$memory_gib_seconds[ind_clean_mem]
+missingdata$date <- data_agg_sum_trans_log[missingdata$index, "date"]
+
+data_agg_sum_trans_log_prep <- GetTimeInfo(data_agg_sum_trans_log)
+data_agg_sum_trans_log_prep$memory_gib_seconds[ind_clean_mem] <- NA
+data_agg_sum_trans_log_prep$gcu_seconds[ind_clean_gcu] <- NA
+
+# Select the features except cell, date, and gpu to impute.
+data_agg_sum_trans_log_prep_mat <-as.matrix(cbind(data_agg_sum_trans_log_prep[, 5], data_agg_sum_trans_log_prep[, 6]))
+set.seed(123)
+imp_prep <- mice::mice(data_agg_sum_trans_log_prep[,1:6], meth="cart")
+imp_prep2 <- mice::complete(imp_prep)
+str(imp_prep2)
+data_agg_sum_trans_log_prep[, 1:6] <- imp_prep2
+
+# Check the results of imputation.
+plot(data_agg_sum_trans_log$date, data_agg_sum_trans_log$gcu_seconds, type = "l")
+lines(data_agg_sum_trans_log_prep$date, data_agg_sum_trans_log_prep$gcu_seconds, col = 2)
+plot(data_agg_sum_trans_log$date,data_agg_sum_trans_log$memory_gib_seconds, col = 1, type = "l")
+lines(data_agg_sum_trans_log_prep$date, data_agg_sum_trans_log_prep$memory_gib_seconds, col = 2)
+
+# Save results
+saveRDS(data_agg_sum_trans_log_prep, "data_agg_sum_trans_log_prep_rf.rds")
+data_agg_sum_trans_log_prep <- readRDS("data_agg_sum_trans_log_prep_rf.rds")
+
+### De-seasonality ###
+# Transformed data de-seasonality
+data_agg_sum_trans_log <- data_agg_sum_trans_log_prep[,5:9]
+data_agg_trans_sum_plt <- GetTimeInfo(data_agg_sum_trans_log)
+stl(data_agg_trans_sum_plt$memory_gib_seconds, s.window="periodic")
+decompose(data_agg_trans_sum_plt$memory_gib_seconds, type = c("multiplicative"), filter = NULL)
+decompose(data_agg_trans_sum_plt$gcu_seconds, type = c("multiplicative"), filter = NULL)
+
+stl(data_agg_sum$gcu_seconds, s.window="periodic")
+decompose(data_agg_sum$gcu_seconds, type = c("multiplicative"), filter = NULL)
```

Comment:
These decomposition approaches don't work, right? You can delete them and just add a comment to this section along the lines of "Tried seasonal decomposition via stl and decompose. Neither picked up the seasonality present in the data."